dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListener.java

```diff
@@ -372,21 +370,20 @@ protected void notifyAddressChanged() {
      * Since this listener is shared among interfaces, destroy this listener only when all interface listener are unsubscribed
      */
     public synchronized void destroy() {
-        if (!destroyed.get()) {
+        if (!destroyed && CollectionUtils.isEmptyMap(listeners)) {
             if (CollectionUtils.isEmptyMap(listeners)) {
```

Comment:
`CollectionUtils.isEmptyMap(listeners)` is roundabout

Reply:
done